### DISCUSSION TITLE
DirectML Support

### DISCUSSION THREAD

**Discussion by waylaa** [xamarin/TorchSharp#1228] [Ideas]
Are there any plans for having a DirectML backend?

**Comment by NiklasGustafsson** [comment]
That would be something for the libtorch project to first handle, and then we can add the API-level support needed... TorchSharp is merely a thin layer on top of libtorch.

That said, it's a good idea!

**Reply by waylaa** [reply]
There's [torch-directml](https://pypi.org/project/torch-directml/) but not any libtorch equivalent which is a bummer as also shown here https://github.com/microsoft/DirectML/issues/247
